augur/subsample.py

```diff
@@ -232,6 +232,58 @@ def run(args: argparse.Namespace) -> None:
                 sample.remove_output_strains()
 
 
+def get_referenced_files(
+    config_file: str,
+    config_section: Optional[List[str]] = None,
+    search_paths: Optional[List[str]] = None,
+) -> Set[str]:
+    """Get the files referenced in a subsample config file.
```

Comment:
_non-blocking_

This seems likely to fall out of date with the behaviour of `augur subsample` since it's not used directly. It would be better if `augur subsample`'s own code and this function shared more of a code path. Perhaps file paths should be resolved as an early transformation of `config`, rather than transformed one by one as `Sample`s are constructed?

Reply:
I ended up doing this with a function [`_resolve_filepaths()`](https://github.com/nextstrain/augur/blob/25c4448a542290d905b4973d6acc2d7f4e3e6c20/augur/subsample.py#L347-L395). It recursively walks through the config/schema and might be slightly hard to follow, but I think it's a decent way to share the same code path between `augur subsample` and `get_referenced_files()`.